hudi-client/hudi-client-common/src/main/java/org/apache/hudi/keygen/ComplexAvroKeyGenerator.java

```diff
@@ -43,9 +45,16 @@ public ComplexAvroKeyGenerator(TypedProperties props) {
         .map(String::trim)
         .filter(s -> !s.isEmpty())
         .collect(Collectors.toList());
-    this.recordKeyFunction = ConfigUtils.getBooleanWithAltKeys(props, COMPLEX_KEYGEN_OLD_ENCODING)
-        ? record -> KeyGenUtils.getRecordKey(record, getRecordKeyFieldNames(), isConsistentLogicalTimestampEnabled())
-        : this::getRecordKeyWithoutKeyFieldNameOnSingleKeyField;
+    // Get table version
+    Integer tableVersionCode = ConfigUtils.getIntWithAltKeys(props, WRITE_TABLE_VERSION);
```

Comment:
we can write a util method in `ConfigUtils.getWriteTableVersion(props)` to return the props directly.

Reply:
Do we really require this?